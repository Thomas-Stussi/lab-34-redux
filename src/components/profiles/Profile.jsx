/* eslint-disable keyword-spacing */
/* eslint-disable max-len */
import React from 'react';
import { useSelector } from 'react-redux';

const Profile = () => {
  const user = useSelector(state => state.user);
  const loading = useSelector(state => state.loading);

  if (loading) {
    return <h1>Loading...</h1>;

  } else if (user === {}) {
    const repoElements = user.userRepos.map(repo => (
      <li key={repo.name}>
        <p>Name: {repo.name}</p>
        <a href={repo.url}>Link: {repo.url}</a>
      </li>
    ));

    return (
      <>
        <h1>{user.userProfile.name}</h1>
        <a href={user.userProfile.url}>Link: {user.userProfile.url}</a>
        <p>Follower Count: {user.userProfile.followers}</p>
        <p>Following {user.userProfile.followers} Users</p>
        <ul>
          Repositories:
          {repoElements}
        </ul>
      </>
    );
  } else { return <p>Please Make a Search</p>; }
};

export default Profile;
