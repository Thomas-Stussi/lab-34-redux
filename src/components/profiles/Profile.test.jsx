import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Profile from './Profile';
import { Provider } from 'react-redux';
import store from '../../store';

describe('Profile component', () => {
  afterEach(() => cleanup());
  

  it('renders Profile', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Profile />
      </Provider>);
    expect(asFragment()).toMatchSnapshot();
  });
});
