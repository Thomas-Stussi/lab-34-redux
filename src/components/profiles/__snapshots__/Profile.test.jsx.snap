// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component renders Profile 1`] = `
<DocumentFragment>
  <p>
    Please Make a Search
  </p>
</DocumentFragment>
`;
